### ISSUE TITLE
👾 [Bug Report] when studying a profession with an NPC, crash

### ISSUE THREAD

**Issue by schnek** [AscEmu/AscEmu#1099]
**Description**: 
in the study of riding, professions  or learning any spell - crash

**Steps to reproduce the problem**: 
1 go to the riding trainer and learn any spell
1 learn a spell from your class
2 crash

**Expansion version**: 

Specify version - is required:
- Patch 2.4.3 - TBC
- Patch 3.3.5 - WoTLK
- Patch 4.3.4 - Cataclysm
- Patch 5.4.8 - Mop

**AscEmu hash/commit**: 

latest develop

**Operating system**: 

Specify the operating system - is required:
- Platform: Windows

**Comment by Appled** [comment]
Does it crash on all versions?

**Comment by schnek** [comment]
yes for me in all versions there is such a problem 🤨

**Comment by Appled** [comment]
Btw does it crash on master branch?

**Comment by schnek** [comment]
not, i tested with develop line: https://github.com/AscEmu/AscEmu/blob/develop/src/world/Spell/SpellInfo.hpp#L134

**Comment by aaron02** [comment]
![image](https://github.com/AscEmu/AscEmu/assets/1266630/973eb0fc-0555-4748-9d90-08cf606fa923)
doing this in NPCHandler fixes the issue, but @Zyres should look at it if its right ^^

**Comment by schnek** [comment]
```
    TrainerSpell const* trainerSpell = nullptr;
    std::vector<TrainerSpell>::const_iterator itr;
    std::vector<TrainerSpell> itr = sObjectMgr.getTrainserSpellSetById(trainet->spell_id);
    
    for (itr = itr.begin(); itr != itr.end(); itr++)
    {
        if ((itr->castSpell && itr->castSpell->getId() == srlPacket->spellId) ||
            (itr->learnSpell && itr->learnSpell->getId() == srlPacket->spellId))
        {
            trainerSpell = &(*itr);
            break;
        }
    }
```
yes this works as it should and fixes the problem

**Comment by Zyres** [comment]
@aaron02 looks good <3

**Comment by schnek** [comment]
fix 761ef69c43f2326327748707a36836f5b7b434fe
